java/vector/src/main/java/org/apache/arrow/vector/NullableUInt2Vector.java

```diff
@@ -0,0 +1,327 @@
+/*******************************************************************************
+
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.apache.arrow.vector;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.complex.impl.UInt2ReaderImpl;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.holders.UInt2Holder;
+import org.apache.arrow.vector.holders.NullableUInt2Holder;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.TransferPair;
+import org.slf4j.Logger;
+
+/**
+ * NullableUInt2Vector implements a fixed width (2 bytes) vector of
+ * integer values which could be null. A validity buffer (bit vector) is
+ * maintained to track which elements in the vector are null.
+ */
+public class NullableUInt2Vector extends BaseNullableFixedWidthVector {
```

Comment:
What's the difference between this and NullableSmallIntVector?

Reply:
I actually think we should remove all the UInt vectors. This was a task we talked about doing a year ago but just didn't get it done. They aren't complete and aren't tested.

Reply:
Oh sweet. Just curious what's the initial intention for those?

Reply:
Yes that is probably a good idea. I have been wondering what is the difference between Int4Vector and UInt4Vector since the latter really doesn't implement unsigned semantics.

Reply:
+1 for removing them

Reply:
Well, the problem with removing them is that other implementations may send unsigned integer data. We've already been integration testing this: https://github.com/apache/arrow/blob/master/integration/integration_test.py#L745

Reply:
I think it's fine to mark the unsigned int vectors in Java "buggy" but they will have to get implemented properly at some point

Reply:
So I guess we are not removing them ?

Keeping them as is and will open JIRA(s) for correctly implementing and testing UINT1, UINT2, UINT4, UINT8. I can open JIRAs

Reply:
OK, great, thank you =)